From 35e2a7332239b143a1e9331bad10b2a908c483b4 Mon Sep 17 00:00:00 2001
From: "Andrew F. Davis" <afd@ti.com>
Date: Fri, 9 Sep 2016 09:54:57 -0500
Subject: [PATCH 4/6] plat-ti: Add DRA72x platform flavor

The DRA72x EVM is based on the single core variant of the DRA7xx
platform. Add this EVM here. We also change the flavor name to
be the full EVM name as changes to the EVM may change values
for OPTEE, for instance with the revision C of this EVM the
memory size and default OPTEE load address change.

Signed-off-by: Andrew F. Davis <afd@ti.com>
---
 core/arch/arm/plat-ti/conf.mk           |  2 +-
 core/arch/arm/plat-ti/platform_config.h | 27 +++++++++++++++++++++++++--
 2 files changed, 26 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-ti/conf.mk b/core/arch/arm/plat-ti/conf.mk
index 7ac154f..b6ee1d3 100644
--- a/core/arch/arm/plat-ti/conf.mk
+++ b/core/arch/arm/plat-ti/conf.mk
@@ -1,4 +1,4 @@
-PLATFORM_FLAVOR ?= dra7xx
+PLATFORM_FLAVOR ?= dra7_evm
 
 # 32-bit flags
 arm32-platform-cpuarch		:= cortex-a15
diff --git a/core/arch/arm/plat-ti/platform_config.h b/core/arch/arm/plat-ti/platform_config.h
index ea456db..fdfe3ac 100644
--- a/core/arch/arm/plat-ti/platform_config.h
+++ b/core/arch/arm/plat-ti/platform_config.h
@@ -28,7 +28,7 @@
 #ifndef PLATFORM_CONFIG_H
 #define PLATFORM_CONFIG_H
 
-#if defined(PLATFORM_FLAVOR_dra7xx)
+#if defined(PLATFORM_FLAVOR_dra7_evm)
 
 #define DRAM0_BASE		0xde900000
 #define DRAM0_SIZE		0x01700000
@@ -37,7 +37,7 @@
 #error Pager not supported on this platform
 #endif /*CFG_WITH_PAGER*/
 
-/* Location of protected DDR on the DRA7xx platform */
+/* Location of protected DDR on the DRA7xx EVM */
 #define TZDRAM_BASE		0xde900000
 #define TZDRAM_SIZE		0x01300000
 
@@ -51,6 +51,29 @@
 #define GIC_BASE            0x48211000
 #define SECRAM_BASE         0x40200000
 
+#elif defined(PLATFORM_FLAVOR_dra72_evm_revc)
+
+#define DRAM0_BASE		0xfd900000
+#define DRAM0_SIZE		0x01700000
+
+#ifdef CFG_WITH_PAGER
+#error Pager not supported on this platform
+#endif /*CFG_WITH_PAGER*/
+
+/* Location of protected DDR on the DRA72x EVM rev. C */
+#define TZDRAM_BASE		0xfd900000
+#define TZDRAM_SIZE		0x01300000
+
+#define CFG_TEE_CORE_NB_CORE	1
+
+/* UART1 */
+#define CONSOLE_UART_BASE		0x4806A000
+#define CONSOLE_UART_CLK_IN_HZ	48000000
+#define UART_BAUDRATE			115200
+
+#define GIC_BASE            0x48211000
+#define SECRAM_BASE         0x40200000
+
 #else
 #error "Unknown platform flavor"
 #endif
-- 
2.7.4

